client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -530,6 +530,8 @@ void reconnect() {
                          try {
                              connection.send(cmd);
                          } catch (ConnectionFailedException e1) {
+                             state.failConnection(e1); // This needs to be invoked here because call to failConnection
```

Comment:
I'd rather add this comment between lines 532 and 533 as like this the indentation looks odd. Also, could we add a comment to the `SegmentOutputStreamImpl.State` class? The comment should explain that there is a potential race between `newConnection` and `failedConnection`, which we can't completely avoid because we have conciously made the decision of not putting calls to the connection under a lock

Reply:
done, added additional comments.